src/Lists.js

```diff
@@ -0,0 +1,264 @@
+        setNewListDialogOpen(true);
+    };
+
+    const handleNewListDialogClose = () => {
+        setNewListDialogOpen(false);
+        setShowAlert(false);
+        setListName("")
+    };
+
+    const handleSetListName = e => {
+        setListName(e.target.value)
+        if (e.target.value.length>0) {
+            setShowAlert(false);
+        }else{
+            setShowAlert(true);
+        }
+    }
+
+    function submitNewList(){
+        const listId = generateUniqueID()
+        setDoc(doc(db, "Lists", listId), {
+            id: listId,
+            name: listName,
+            showCompleted: true,
+            sortBy: "priority",
+            sortDir: "desc",
+            created: serverTimestamp(),
+        })
+        if (lists.length===0) {
+            setTabId(listId)
+        }
+    }
+
+    function onSubmitNewList(e) {
+        if (listName.length===0) {
+            setShowAlert(true)
+            setListName("")
+        }else{
+            submitNewList()
+            handleNewListDialogClose();
+            setListName("")
+        }
+    }
+
+    const handleClickMenu = (event) => {
+        setAnchorElMenu(event.currentTarget);
+    };
+
+    const handleCloseMenu = () => {
+        setAnchorElMenu(null);
+    };
+
+    const handleMenuAction = (e, option) => {
+        if (option === "Start a New List") {
+            handleNewListDialogOpen()
+        }
+        setAnchorElMenu(null);
+    };
+
+    function setTab(deletedId) {
+        if (lists[0].id === deletedId && lists.length === 1) {
+            setTabId('none')
+        }else if (lists[0].id === deletedId && lists.length !== 1) {
+            setTabId(lists[1].id)
+        }else {
+            setTabId(lists[0].id)
+        }
+    }
+
+    useEffect(() => {
+        if (lists && lists.length!==0) {
+            setTabId(prevId => prevId==='none' ? lists[0].id : prevId)
+        }
+    }, [lists])
+    
+    if (loading) {
+        return (
+            <p>Loading</p>
+        )
+    }else if (error) {
+        return (
+            <p>Error: {JSON.stringify(error)}</p>
+        )
+    }else if (lists.length===0) {
+        return (
+            <>
+            <Button onClick={handleNewListDialogOpen} variant='outlined' sx={{textTransform:'none', marginRight:2, color:'primary.dark'}}>Start a New List</Button>
+            <Dialog open={newListDialogOpen} onClose={handleNewListDialogClose}>
+                <DialogTitle aria-label="Create a New List. Please enter the name of the new list below.">Create a New List</DialogTitle>
+                <DialogContent>
+                <DialogContentText>
+                    Please enter the name of the new list below.
+                </DialogContentText>
+                <TextField
+                    autoFocus
+                    margin="dense"
+                    id="list-name"
+                    aria-label={!showAlert ? `List name` : `Please enter a non-empty name for the new list`}
```

Comment:
I imagine that users of screenreaders want fairly tight labels (because they can only list so quickly).  I'd just have "Name for new list". Or, wrap line 152 in a <label> whose 'for' is this text field.